src/dune_rules/odoc.ml

```diff
@@ -448,10 +506,48 @@ let setup_toplevel_index_rule sctx =
   </body>
 </html>|}
       Paths.odoc_support_dirname
-      list_items
-  in
+      (html_list_items t)
+  ;;
+
+  let string_to_json s = `String s
+  let list_to_json ~f l = `List (List.map ~f l)
+
+  let option_to_json ~f = function
+    | None -> `Null
+    | Some x -> f x
+  ;;
+
+  let item_to_json { name; version; link } =
+    `Assoc
+      [ "name", string_to_json name
+      ; "version", option_to_json ~f:string_to_json version
+      ; "link", string_to_json link
+      ]
+  ;;
+
+  (** This format is public API. *)
```

Comment:
There's no formal schema for this, but this is printed in full in a test to avoid changing accidentally. Would you prefer a schema in this docstring?

Re #7840, I think that the question is what are you doing on the HTML side of things? If the toplevel index is modified, I think that the JSON one should too. But we don't have important versioning constraints here (the schema of the output depends on the version of dune itself).

Reply:
OK. I see ne problem then, this JSON file is going to disappear in a few releases then.